src/Laravel/routes/api.php

```diff
@@ -40,13 +40,15 @@
                 }
 
                 /* @var HttpOperation $operation */
-                Route::addRoute($operation->getMethod(), Str::replace('{._format}', '{_format?}', $operation->getUriTemplate()), ApiPlatformController::class)
-                    ->prefix($operation->getRoutePrefix())
-                    ->middleware(ApiPlatformMiddleware::class.':'.$operation->getName())
```

Comment:
This can return an array if a number of things are passed to it which causes the next chained methods to fail. Pushing up into a group ensures we always return an instance of `Route` for the chained methods.

Reply:
I found what's wrong, you can not group here as an operation is only one route.

Reply:
Also the `:$arg` hack is to get the `operationName` inside our Middleware.

Reply:
So we are still running into the issue of when passing an arg to `middleware`, if it is null, it will return an array. So `operation->getMiddleware()` often returns null, and causes a `Call to a member function where() on array`. Placing them in a single middleware wrapped by an array ensures it is never null. But again, the route prefix is still not getting applied for some reason.

Reply:
Can't you apply `middleware()` only when it's defined?